chandra_aca/fov_plot.py

```diff
@@ -0,0 +1,309 @@
+    :param attitude: A Quaternion compatible attitude for the pointing
+    :param stars: astropy table compatible set of agasc records of stars
+    :param title: string to be used as suptitle for the figure
+    :param red_mag_lim: faint limit for field star plotting.
+    :param quad_bound: boolean, plot inner quadrant boundaries
+    :param grid: boolean, plot axis grid
+    :param bad_stars: boolean mask on 'stars' of those that don't meet minimum requirements
+                      to be selected as acq stars
+    :returns: matplotlib figure
+    """
+    fig = plt.figure(figsize=(5.325, 5.325))
+    ax = fig.add_subplot(1, 1, 1)
+    plt.subplots_adjust(top=0.95)
+    ax.set_aspect('equal')
+
+    # plot the box and set the labels
+    plt.xlim(2900, -2900)
+    plt.ylim(-2900, 2900)
+    b1hw = 2560
+    box1 = plt.Rectangle((b1hw, -b1hw), -2 * b1hw, 2 * b1hw,
+                         fill=False)
+    ax.add_patch(box1)
+    b2w = 2600
+    box2 = plt.Rectangle((b2w, -b1hw), -4 + -2 * b2w, 2 * b1hw,
+                         fill=False)
+    ax.add_patch(box2)
+
+    ax.scatter([-2700, -2700, -2700, -2700, -2700],
+               [2400, 2100, 1800, 1500, 1200],
+               c='orange', edgecolors='none',
+               s=symsize(np.array([10.0, 9.0, 8.0, 7.0, 6.0])))
+
+    [l.set_rotation(90) for l in ax.get_yticklabels()]
+    ax.grid(grid)
+    ax.set_ylabel("Zag (arcsec)")
+    ax.set_xlabel("Yag (arcsec)")
+
+    if quad_bound:
+        pix_range = np.linspace(-510, 510, 50)
+        minus_half_pix = -0.5 * np.ones_like(pix_range)
+        # plot the row = -0.5 line
+        yag, zag = pixels_to_yagzag(minus_half_pix, pix_range)
+        ax.plot(yag, zag, color='magenta', alpha=.4)
+        # plot the col = -0.5 line
+        yag, zag = pixels_to_yagzag(pix_range, minus_half_pix)
+        ax.plot(yag, zag, color='magenta', alpha=.4)
+
+    # plot field if present
+    if stars is not None:
+        if attitude is None:
+            raise ValueError("Must supply attitude to plot field stars")
+        _plot_field_stars(ax, stars, Quaternion.Quat(attitude),
+                          bad_stars=bad_stars, red_mag_lim=red_mag_lim)
+    # plot starcheck catalog
+    if catalog is not None:
+        _plot_catalog_items(ax, catalog)
+    if title is not None:
+        fig.suptitle(title, fontsize='small')
+    return fig
+
+
+def bad_acq_stars(stars):
+    """
+    Return mask of 'bad' stars, by evaluating AGASC star parameters.
+    :param stars: astropy table-compatible set of agasc records of stars
+    :returns: boolean mask true for 'bad' stars
+    """
+    return ((stars['CLASS'] != 0) |
+            (stars['MAG_ACA_ERR'] > 100) |
+            (stars['POS_ERR'] > 3000) |
+            (stars['ASPQ1'] > 0) |
+            (stars['ASPQ2'] > 0) |
+            (stars['ASPQ3'] > 999) |
+            (stars['VAR'] > -9999))
+
+
+def plot_catalog(ra, dec, roll, catalog, starcat_time=DateTime(),
```

Comment:
Putting things like `starcat_time=DateTime()` in the function definition is almost always a bad idea.  In this particular case it will use the DateTime() at function import, not when it is run.